Assets/Scripts/NPCs/BasicNurse.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using NPCs.Components;
+using UnityEngine;
+
+namespace NPCs
+{
+    [RequireComponent(typeof(Chaser))]
+    [RequireComponent(typeof(Responder))]
+    [RequireComponent(typeof(Searcher))]
+    [RequireComponent(typeof(Patroller))]
+    [RequireComponent(typeof(Waiter))]
+    [RequireComponent(typeof(Animator))]
+    public class BasicNurse : BaseAi
+    {
+        private static readonly HashSet<string> States = new HashSet<string>()
+        {
+            "Chasing",
+            "Responding",
+            "Searching",
+            "Patrolling",
+        };
+
+        [HideInInspector] public Chaser chaser;
+        [HideInInspector] public Responder responder;
+        [HideInInspector] public Searcher searcher;
+        [HideInInspector] public Patroller patroller;
+        [HideInInspector] public Waiter waiter;
+
+        public int numSearches = 3;
+        private int curNumSearches = 0;
+        
+        private GameObject questionMark;
+        private Animator animator;
+
+        private new void Awake()
+        {
+            base.Awake();
+
+            chaser = Utils.GetRequiredComponent<Chaser>(this);
+            responder = Utils.GetRequiredComponent<Responder>(this);
+            searcher = Utils.GetRequiredComponent<Searcher>(this);
+            patroller = Utils.GetRequiredComponent<Patroller>(this);
+            waiter = Utils.GetRequiredComponent<Waiter>(this);
+            animator = Utils.GetRequiredComponentInChildren<Animator>(this);
+
+            questionMark = transform.Find("QuestionMark").gameObject;
+            
+            SetState(currentState);
+        }
+
+        public override void SetState(string newState)
+        {
+            switch (newState)
+            {
+                case "Searching":
+                    agent.isStopped = false;
+                    agent.speed = searcher.movementSpeed;
+                    questionMark.SetActive(true);
+                    break;
+                case "Responding":
+                    agent.isStopped = false;
+                    agent.speed = responder.movementSpeed;
+                    questionMark.SetActive(true);
+                    break;
+                case "Chasing":
+                    agent.isStopped = false;
+                    agent.speed = chaser.movementSpeed;
+                    questionMark.SetActive(false);
+                    break;
+                case "Patrolling":
+                    agent.isStopped = false;
+                    agent.speed = patroller.movementSpeed;
+                    questionMark.SetActive(false);
+                    break;
+                default:
+                    questionMark.SetActive(false);
+                    throw new UnityException("Invalid state name passed to " + GetType().Name);
+            }
```

Comment:
Hey Finn!

I want to add a few FMOD params (float) for when you get caught so that the music dynamically changes when the nurse is searching and chasing. Could we chat about how we can put that in during your refactor?

Reply:
Yeah no problem, once this PR is merged into master I can help you on a different branch to add FMOD stuff into the refactored AI system!